app/src/lib/api.ts

```diff
@@ -279,12 +290,18 @@ export class API {
     }
   }
 
-  /** Fetch the logged in account. */
-  public async fetchAccount(): Promise<IAPIUser> {
+  /** Fetch the logged in account and the current scopes associated with the token. */
+  public async fetchAccount(): Promise<UserWithTokenScopes> {
     try {
       const response = await this.request('GET', 'user')
-      const result = await parsedResponse<IAPIUser>(response)
-      return result
+      const user = await parsedResponse<IAPIUser>(response)
+      const scopeHeader = response.headers.get('X-OAuth-Scopes')
+      const scopes =
+        scopeHeader == null
+          ? []
+          : scopeHeader.split(',').map(scope => scope.trim())
+
+      return { user, scopes }
```

Comment:
Is it so you wont have to explicitly annotate the return value?

Reply:
Nah, I don't think this behaviour is different between interfaces and plain type signatures